apps/snitch_core/lib/core/data/model/product.ex

```diff
@@ -73,13 +80,13 @@ defmodule Snitch.Data.Model.Product do
     |> where([p], p.state == "active" and p.id not in ^child_product_ids)
   end
 
-  def get_product_with_present_variants(product) do
+  def get_product_with_required_variants(product, states \\ [0, 1, 2]) do
     query =
       Product
       |> join(:left, [p], v in Variation, v.parent_product_id == ^product.id)
```

Comment:
We also have variants assoc. That can be used here.

```elixir
product.variants
|> where(
        [p, v],
        p.state != "deleted" and p.id == v.child_product_id
        p.state in ^states and p.id == v.child_product_id
      )
```

Reply:
@jyotigautam use states `:active`, `:draft`, `:in_active` instead of ids.